### PR TITLE
Added a retry with HttpClient to failed requests, and added the response headers in base64 to the exception

### PR DESCRIPTION
**Pull request by tomer-amir-apiiro**
This is an attempt to get more info for LIM-948

### CODE REVIEW

File: Atlassian.Jira/Remote/JiraRestClient.cs [hunk 1]
```diff
@@ -195,7 +198,8 @@ private JToken GetValidJsonFromResponse(IRestRequest request, IRestResponse resp
 
             if (!string.IsNullOrEmpty(response.ErrorMessage))
             {
-                throw new InvalidOperationException($"Error Message: {response.ErrorMessage}\nContent: {content}\nCode: {response.StatusCode}", response.ErrorException);
+                var headers = PrintResponseAsync(request).Result;
```
Comment (liiri) [inline Atlassian.Jira/Remote/JiraRestClient.cs]: Never use .Result without await first

File: Atlassian.Jira/Remote/JiraRestClient.cs [hunk 2]
```diff
@@ -242,5 +246,43 @@ private JToken GetValidJsonFromResponse(IRestRequest request, IRestResponse resp
...
+                client.DefaultRequestHeaders.Add("Connection", "Keep-Alive");
+                client.DefaultRequestHeaders.Add("Accept", "application/json, text/json, text/x-json, text/javascript, application/xml, text/xml");
+                client.DefaultRequestHeaders.Add("User-Agent", "RestSharp/106.11.7.0");
+                var response = await client.GetAsync($"{_restClient.BaseUrl}{request.Resource}");
+                var headers = response.Headers;
+                return $"Retry request attempt result = {Convert.ToBase64String(Encoding.ASCII.GetBytes(headers.ToString()))}";
+            }
+            catch (Exception exception)
+            {
+                Trace.WriteLine("Failed to retry request");
```
Comment (liiri) [inline Atlassian.Jira/Remote/JiraRestClient.cs]: Why not return this message and the ex details?

File: Atlassian.Jira/Remote/JiraRestClient.cs (hunk 1 above)
Comment (tomer-amir-apiiro) [reply Atlassian.Jira/Remote/JiraRestClient.cs]: Thought `.Result` will block... My mistake...
Will this work?
```csharp
if (!string.IsNullOrEmpty(response.ErrorMessage))
{
    var task = PrintResponseAsync(request);
    task.Wait();
    var headers = task.Result;
    throw new InvalidOperationException($"Error Message: {response.ErrorMessage}\nContent: {content}\nCode: {response.StatusCode}\nHeaders: {headers}", response.ErrorException);
}
```

Never called an async method from a synced method

File: Atlassian.Jira/Remote/JiraRestClient.cs (hunk 2 above)
Comment (tomer-amir-apiiro) [reply Atlassian.Jira/Remote/JiraRestClient.cs]: didn't want to spam, but sure, adding 👍

File: Atlassian.Jira/Remote/JiraRestClient.cs (hunk 1 above)
Comment (liiri) [reply Atlassian.Jira/Remote/JiraRestClient.cs]: Why not 
var msg = await task;

Comment (tomer-amir-apiiro) [reply Atlassian.Jira/Remote/JiraRestClient.cs]: this method is not async, and I think turning it to async might have a ripple effect, but I can try if you think it's important enough

Comment (tomer-amir-apiiro) [reply Atlassian.Jira/Remote/JiraRestClient.cs]: Spoke with @YonatanEldar and we think that this:
```csharp
string headers = null;
try
{
    headers = PrintResponseAsync(request).Result;
}
catch
{
    // ignored
}

throw new InvalidOperationException(
    $"Error Message: {response.ErrorMessage}\nContent: {content}\nCode: {response.StatusCode}\nHeaders: {headers}",
    response.ErrorException);
```
Should be good enough...
WDYT?

Comment (liiri) [reply Atlassian.Jira/Remote/JiraRestClient.cs]: trusting you that the ripple effect is too big to be worth it

**Review by liiri — APPROVED** [review]

File: Atlassian.Jira/Remote/JiraRestClient.cs (hunk 1 above)
Comment (tomer-amir-apiiro) [reply Atlassian.Jira/Remote/JiraRestClient.cs]: 😄

**Pull request merged**